runtime/connectors/blob/blobhandler.go

```diff
@@ -0,0 +1,82 @@
+package blob
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/rilldata/rill/runtime/pkg/fileutil"
+	"gocloud.dev/blob"
+	"golang.org/x/sync/errgroup"
+)
+
+// increasing this limit can increase speed ingestion
+// but may increase bottleneck at duckdb or network/db IO
+// set without any benchamarks
+const concurrentBlobDownloadLimit = 8
+
+type BlobHandler struct {
+	bucket *blob.Bucket
+	prefix string
+	path   string
+
+	BlobType   BlobType
+	FileNames  []string
+	LocalPaths []string
+	TempDir    string
+}
+
+func (b *BlobHandler) Close() {
+	b.bucket.Close()
+	os.RemoveAll(b.TempDir)
+}
+
+// object path is relative to bucket
+func (b *BlobHandler) DownloadObject(ctx context.Context, objpath string) (string, error) {
+	if b.BlobType == File {
+		return fmt.Sprintf("%s/%s", b.path, objpath), nil
+	}
```

Comment:
As stated elsewhere, let's remove file handling from here. Too much of a special case to be worth doing here.

Reply:
Then you can also remove the `BlobType` struct